index/scorch/scorch.go

```diff
@@ -88,14 +88,51 @@ type Scorch struct {
 	spatialPlugin index.SpatialAnalyzerPlugin
 }
 
-// AsyncPanicError is passed to scorch asyncErrorHandler when panic occurs in scorch background process
-type AsyncPanicError struct {
-	Source string
-	Path   string
+const (
+	SourceIntroducer = "introducer"
+	SourcePersister  = "persister"
+	SourceMerger     = "merger"
+)
```

Comment:
Quick nit :)

Let's declare each of these consts in the files they're pertinent to.
Is there a reason these need to be exported? Simply `const introducer = "introducer"` etc. would suffice?